app/src/main/java/com/kickstarter/libs/loadmore/ApolloPaginate.kt

```diff
@@ -0,0 +1,253 @@
+            this.envelopeToListOfData = envelopeToListOfData
+            return this
+        }
+
+        /**
+         * [Required] A function that takes a `Params` and performs the associated network request
+         * and returns an `Observable<Envelope>`
+         </Envelope> */
+        fun loadWithParams(loadWithParams: Func1<Pair<Params, String?>, Observable<Envelope>>): Builder<Data, Envelope, Params> {
+            this.loadWithParams = loadWithParams
+            return this
+        }
+
+        /**
+         * [Optional] Function to transform every page of data that is loaded.
+         */
+        fun pageTransformation(pageTransformation: Func1<List<Data>, List<Data>>): Builder<Data, Envelope, Params> {
+            this.pageTransformation = pageTransformation
+            return this
+        }
+
+        /**
+         * [Optional] Determines if the list of loaded data is cleared when starting over from the first page.
+         */
+        fun clearWhenStartingOver(clearWhenStartingOver: Boolean): Builder<Data, Envelope, Params> {
+            this.clearWhenStartingOver = clearWhenStartingOver
+            return this
+        }
+
+        /**
+         * [Optional] Determines how two lists are concatenated together while paginating. A regular `ListUtils::concat` is probably
+         * sufficient, but sometimes you may want `ListUtils::concatDistinct`
+         */
+        fun concater(concater: Func2<List<Data>?, List<Data>?, List<Data>?>): Builder<Data, Envelope, Params> {
+            this.concater = concater
+            return this
+        }
+
+        /**
+         * [Optional] Determines if the list of loaded data is should be distinct until changed.
+         */
+        fun distinctUntilChanged(distinctUntilChanged: Boolean): Builder<Data, Envelope, Params> {
+            this.distinctUntilChanged = distinctUntilChanged
+            return this
+        }
+
+        /**
+         * [Optional] Determines if the list of loaded data is should be distinct until changed.
+         */
+        fun isReversed(distinctUntilChanged: Boolean): Builder<Data, Envelope, Params> {
+            this.isReversed = isReversed
+            return this
+        }
+
+        @Throws(RuntimeException::class)
+        fun build(): ApolloPaginate<Data, Envelope, Params> {
+            // Early error when required field is not set
+            if (nextPage == null) {
+                throw RuntimeException("`nextPage` is required")
+            }
+            if (envelopeToListOfData == null) {
+                throw RuntimeException("`envelopeToListOfData` is required")
+            }
+            if (loadWithParams == null) {
+                throw RuntimeException("`loadWithParams` is required")
+            }
+
+            // Default params for optional fields
+            if (startOverWith == null) {
+                startOverWith = Observable.just(null)
+            }
+
+            return ApolloPaginate(
+                nextPage!!,
```

Comment:
Change the `!!` over `requireNotNull(nextPage)`

Reply:
ok